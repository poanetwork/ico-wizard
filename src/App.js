import React, {Component} from 'react';
import './assets/stylesheets/application.css';
import { Header, Footer, Home, stepOne, stepTwo, stepThree, stepFour, stepFive, Invest } from './components/index'
import { getQueryVariable } from './utils/utils'
import {
  BrowserRouter as Router,
  Route
} from 'react-router-dom'

class App extends Component {
  render() {
    var crowdsaleAddr = getQueryVariable("addr");
    return (
      <Router>
        <div>
          <Header/>
            <Route exact path="/" component={crowdsaleAddr?stepFive:Home}/>
            <Route path="/1" component={stepOne}/>
            <Route path="/2" component={stepTwo}/>
            <Route path="/3" component={stepThree}/>
            <Route path="/4" component={stepFour}/>
            <Route path="/crowdsale" component={stepFive}/>
            <Route path="/invest" component={Invest}/>
          <Footer/>
        </div>
      </Router>
    )
  }
}

export default App;
