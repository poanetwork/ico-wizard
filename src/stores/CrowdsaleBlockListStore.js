import { observable, action } from 'mobx';

class CrowdsaleBlockListStore {

	@observable crowdsaleBlockList;
	
	constructor(crowdsaleBlockList = []) {
    this.crowdsaleBlockList = crowdsaleBlockList;
  }

  @action addCrowdsaleBlock = (crowdsaleBlock) => {
    this.crowdsaleBlockList.push(crowdsaleBlock)
	}
	
	@action setCrowdsaleBlockProperty = (index, property, value) => {
		let newCrowdsaleBlock = {...this.crowdsaleBlockList[index]}
		newCrowdsaleBlock[property] = value
    this.crowdsaleBlockList[index] = newCrowdsaleBlock;
	}
	
	@action removeCrowdsaleBlock = (index) => {
		this.crowdsaleBlockList.splice(index,1)
	}

}

const crowdsaleBlockListStore = new CrowdsaleBlockListStore();

export default crowdsaleBlockListStore;
export { CrowdsaleBlockListStore };
