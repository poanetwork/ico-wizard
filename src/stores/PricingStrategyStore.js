import { observable, action } from 'mobx';

class PricingStrategyStore {

	@observable strategies;
	
	constructor(strategies = []) {
    this.strategies = strategies;
  }

  @action addStrategy = (strategy) => {
    this.strategies.push(strategy)
	}
	
	@action setStrategyProperty = (index, property, value) => {
		let newStrategy = {...this.strategies[index]}
		newStrategy[property] = value
    this.strategies[index] = newStrategy;
	}
	
	@action removeStrategy = (index) => {
		this.strategies.splice(index,1)
	}

}

const pricingStrategyStore = new PricingStrategyStore();

export default pricingStrategyStore;
export { PricingStrategyStore };
