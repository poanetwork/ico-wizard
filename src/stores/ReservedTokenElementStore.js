import { observable, action } from 'mobx';

class ReservedTokensElementStore {

	@observable tokenElements;
	
	constructor(tokenElements = []) {
    this.tokenElements = tokenElements;
  }

  @action addToken = (token) => {
    this.tokenElements.push(token)
	}
	
	@action setTokenProperty = (index, property, value) => {
		let newToken = {...this.tokenElements[index]}
		newToken[property] = value
    this.tokenElements[index] = newToken;
	}
	
	@action removeElement = (index) => {
		this.tokenElements.splice(index,1)
	}


}

const reservedTokensElementStore = new ReservedTokensElementStore();

export default reservedTokensElementStore;
export { ReservedTokensElementStore };
