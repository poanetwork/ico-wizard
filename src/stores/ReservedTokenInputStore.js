import { observable, action } from 'mobx';

class ReservedTokenInputStore {

	@observable dim;
	@observable addr;
	@observable val;

  @action setProperty = (property, value) => {
    this[property] = value
	}

}

const reservedTokenInputStore = new ReservedTokenInputStore();

export default reservedTokenInputStore;
export { ReservedTokenInputStore };
