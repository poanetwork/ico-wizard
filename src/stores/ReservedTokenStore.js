import { observable, action } from 'mobx';

class ReservedTokenStore {

	@observable tokens;
	
	constructor(tokens = []) {
    this.tokens = tokens;
  }

  @action addToken = (token) => {
    this.tokens.push(token)
	}
	
	@action setTokenProperty = (index, property, value) => {
		let newToken = {...this.tokens[index]}
		newToken[property] = value
    this.tokens[index] = newToken;
	}
	
	@action removeToken = (index) => {
		this.tokens.splice(index,1)
	}

}

const reservedTokenStore = new ReservedTokenStore();

export default reservedTokenStore;
export { ReservedTokenStore };
