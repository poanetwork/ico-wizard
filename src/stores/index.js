import contractStore from './ContractStore';
import pricingStrategyStore from './PricingStrategyStore';
import reservedTokenStore from './ReservedTokenStore';
import reservedTokenInputStore from './ReservedTokenInputStore';
import reservedTokenElementStore from './ReservedTokenElementStore';
import stepThreeValidationStore from './StepThreeValidationStore';
import stepTwoValidationStore from './StepTwoValidationStore';
import tierStore from './TierStore';
import tokenStore from './TokenStore';
import web3Store from './Web3Store';
import tierCrowdsaleListStore from './TierCrowdsaleListStore'
import crowdsaleBlockListStore from './CrowdsaleBlockListStore'

export {
    tierCrowdsaleListStore,
    crowdsaleBlockListStore,
    contractStore,
    pricingStrategyStore,
    reservedTokenStore,
    reservedTokenInputStore,
    reservedTokenElementStore,
    stepThreeValidationStore,
    stepTwoValidationStore,
    tierStore,
    tokenStore,
    web3Store
};
