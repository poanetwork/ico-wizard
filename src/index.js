import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import registerServiceWorker from './registerServiceWorker';
import { useStrict } from 'mobx';
import { Provider } from 'mobx-react';
import * as stores from './stores';
useStrict(true);


ReactDOM.render(
  <Provider { ...stores }>
    <App />
  </Provider>, 
document.getElementById('root'));
registerServiceWorker();
