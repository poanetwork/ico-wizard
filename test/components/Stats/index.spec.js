import React from 'react'
import Adapter from 'enzyme-adapter-react-15'
import { Provider } from 'mobx-react'
import { configure } from 'enzyme'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'
import { Stats } from '../../../src/components/Stats/index'
import { web3Store, statsStore } from '../../../src/stores'

configure({ adapter: new Adapter() })

describe('Stats index', () => {
  const stores = {
    web3Store,
    statsStore
  }

  it(`should render Stats`, () => {
    // Given
    const component = renderer.create(
      <Provider {...stores}>
        <MemoryRouter initialEntries={['/stat']}>
          <Stats />
        </MemoryRouter>
      </Provider>
    )

    // When
    const tree = component.toJSON()

    // Then
    expect(tree).toMatchSnapshot()
  })
})
