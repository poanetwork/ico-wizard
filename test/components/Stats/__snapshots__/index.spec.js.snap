// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats index should render Stats 1`] = `
<div>
  <section
    className="lo-MenuBarAndContent"
  >
    <div
      className="mng-ManageNavigation"
    >
      <a
        className="sw-LogoWhite"
        href="/"
      />
      <div
        className="mng-ManageNavigation_Container"
      >
        <div
          className="st-StepItem
        active  "
        >
          <div
            className="st-StepItem_Circle"
          />
          <span
            className="st-StepItem_Text"
          >
            Statistics
          </span>
        </div>
      </div>
    </div>
    <div
      className="st-StepContent"
    >
      <div
        className="st-SectionInfo "
      >
        <div
          className="st-SectionInfo_Icon st-SectionInfo_Icon-step0"
        />
        <div
          className="st-SectionInfo_Text"
        >
          <h1
            className="st-SectionInfo_Title"
          >
            Statistics
          </h1>
          <p
            className="st-SectionInfo_Description"
            dangerouslySetInnerHTML={
              Object {
                "__html": "Useful statistics about the Token Wizard",
              }
            }
          />
        </div>
      </div>
      <div
        className="stats-table"
      >
        <div
          className="stats-table-cell"
        >
          <div
            className="stats-items"
          >
            <div
              className="stats-items-i"
            >
              <p
                className="stats-items-title"
              >
                0
              </p>
              <p
                className="stats-items-description"
              >
                Total crowdsales amount
              </p>
            </div>
          </div>
        </div>
        <div
          className="stats-table-cell"
        >
          <div
            className="stats-items"
          >
            <div
              className="stats-items-i"
            >
              <p
                className="stats-items-title"
              >
                0
              </p>
              <p
                className="stats-items-description"
              >
                Total amount of eth raised
              </p>
            </div>
          </div>
        </div>
        <div
          className="stats-table-cell"
        >
          <div
            className="stats-items"
          >
            <div
              className="stats-items-i"
            >
              <p
                className="stats-items-title"
              >
                0
              </p>
              <p
                className="stats-items-description"
              >
                Total contributors amount
              </p>
            </div>
          </div>
        </div>
        <div
          className="stats-table-cell"
        >
          <div
            className="stats-items"
          >
            <div
              className="stats-items-i"
            >
              <p
                className="stats-items-title"
              >
                0
              </p>
              <p
                className="stats-items-description"
              >
                Max amount of eth raised in one crowdsale
              </p>
            </div>
          </div>
        </div>
      </div>
      <p
        className="title-secondary"
      >
        Minted capped crowdsales statistics
      </p>
      <div
        className="stats-table"
      >
        <div
          className="stats-table-cell"
        >
          <div
            className="stats-items"
          >
            <div
              className="stats-items-i"
            >
              <p
                className="stats-items-title"
              >
                0
              </p>
              <p
                className="stats-items-description"
              >
                Crowdsales amount
              </p>
            </div>
            <div
              className="stats-items-i"
            >
              <p
                className="stats-items-title"
              >
                0
              </p>
              <p
                className="stats-items-description"
              >
                Total amount of eth raised
              </p>
            </div>
            <div
              className="stats-items-i"
            >
              <p
                className="stats-items-title"
              >
                0
              </p>
              <p
                className="stats-items-description"
              >
                Total contributors amount
              </p>
            </div>
          </div>
        </div>
        <div
          className="stats-table-cell"
        >
          <div
            className="stats-items"
          >
            <div
              className="stats-items-i"
            >
              <p
                className="stats-items-title"
              >
                0
              </p>
              <p
                className="stats-items-description"
              >
                Ongoing crowdsales amount
              </p>
            </div>
            <div
              className="stats-items-i"
            >
              <p
                className="stats-items-title"
              >
                0
              </p>
              <p
                className="stats-items-description"
              >
                Future crowdsales amount
              </p>
            </div>
            <div
              className="stats-items-i"
            >
              <p
                className="stats-items-title"
              >
                0
              </p>
              <p
                className="stats-items-description"
              >
                Past crowdsales amount
              </p>
            </div>
          </div>
        </div>
        <div
          className="stats-table-cell"
        >
          <div
            className="stats-items"
          >
            <div
              className="stats-items-i"
            >
              <p
                className="stats-items-title"
              >
                0
              </p>
              <p
                className="stats-items-description"
              >
                Max amount of eth raised in one crowdsale
              </p>
            </div>
            <div
              className="stats-items-i"
            >
              <p
                className="stats-items-title"
              >
                0
              </p>
              <p
                className="stats-items-description"
              >
                % of finalized crowdsales from ended
              </p>
            </div>
            <div
              className="stats-items-i"
            >
              <p
                className="stats-items-title"
              >
                0
              </p>
              <p
                className="stats-items-description"
              >
                % of crowdsales with multiple tiers
              </p>
            </div>
          </div>
        </div>
        <div
          className="stats-table-cell"
        >
          <div
            className="stats-items"
          >
            <div
              className="stats-items-i"
            >
              <p
                className="stats-items-title"
              >
                0
              </p>
              <p
                className="stats-items-description"
              >
                Max tiers amount in one crowdsale
              </p>
            </div>
          </div>
        </div>
      </div>
      <p
        className="title-secondary"
      >
        Dutch auction crowdsales statistics
      </p>
      <div
        className="stats-table"
      >
        <div
          className="stats-table-cell"
        >
          <div
            className="stats-items"
          >
            <div
              className="stats-items-i"
            >
              <p
                className="stats-items-title"
              >
                0
              </p>
              <p
                className="stats-items-description"
              >
                Crowdsales amount
              </p>
            </div>
            <div
              className="stats-items-i"
            >
              <p
                className="stats-items-title"
              >
                0
              </p>
              <p
                className="stats-items-description"
              >
                Total amount of eth raised
              </p>
            </div>
            <div
              className="stats-items-i"
            >
              <p
                className="stats-items-title"
              >
                0
              </p>
              <p
                className="stats-items-description"
              >
                Total contributors amount
              </p>
            </div>
          </div>
        </div>
        <div
          className="stats-table-cell"
        >
          <div
            className="stats-items"
          >
            <div
              className="stats-items-i"
            >
              <p
                className="stats-items-title"
              >
                0
              </p>
              <p
                className="stats-items-description"
              >
                Ongoing crowdsales amount
              </p>
            </div>
            <div
              className="stats-items-i"
            >
              <p
                className="stats-items-title"
              >
                0
              </p>
              <p
                className="stats-items-description"
              >
                Future crowdsales amount
              </p>
            </div>
            <div
              className="stats-items-i"
            >
              <p
                className="stats-items-title"
              >
                0
              </p>
              <p
                className="stats-items-description"
              >
                Past crowdsales amount
              </p>
            </div>
          </div>
        </div>
        <div
          className="stats-table-cell"
        >
          <div
            className="stats-items"
          >
            <div
              className="stats-items-i"
            >
              <p
                className="stats-items-title"
              >
                0
              </p>
              <p
                className="stats-items-description"
              >
                Max amount of eth raised in one crowdsale
              </p>
            </div>
            <div
              className="stats-items-i"
            >
              <p
                className="stats-items-title"
              >
                0
              </p>
              <p
                className="stats-items-description"
              >
                % of finalized crowdsales from ended
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
